### PR TITLE
Streamline how to fake an App Engine environment

### PR DESCRIPTION
**Pull request by jianglai**
Both `DatastoreEntityExtension.PlaceholderEnvironment` and `AppEngineEnvironment` does the same thing, so there is no point having both of them exist. To use `AppEngineEnvionrment` as an autoclosable requires the user to be mindful of where a fake App Engine environment is required. It is better to set this either in the `DatastoreEntityExtension` for tests, or in the worker initializer in Beam. It also makes it easier to remove the fake environment when we are completely datastore free.

Also made a change to how `IdService` allocate Ids in Beam.

This change is [Reviewable](https://reviewable.io/reviews/google/nomulus/1348)

### CODE REVIEW

**Review by weiminyu — CHANGES_REQUESTED** [review]
> Reviewed 8 of 13 files at r1, 2 of 2 files at r2, 3 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1348)* status: 13 of 17 files reviewed, 1 unresolved discussion (waiting on @jianglai and @weiminyu)

___
*[core/src/main/java/google/registry/beam/common/RegistryPipelineWorkerInitializer.java, line 46 at r3](https://reviewable.io/reviews/google/nomulus/1348#-MksA3LS7o1ZD2K6YvY6:-MksA3LTAnyKu385O0Ql:bs4diw7) ([raw file](https://github.com/google/nomulus/blob/cf8b8f6e932b1a124033fad7621dbb7debb9e8d7/core/src/main/java/google/registry/beam/common/RegistryPipelineWorkerInitializer.java#L46)):*
> ```
>  == 
> ```
equals() is favored, at least internally: go/java-practices/enums#equals

**Review by jianglai — COMMENTED** [review]
> Reviewed 13 of 13 files at r1, 2 of 2 files at r2, 3 of 3 files at r3.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1348)* status: 16 of 17 files reviewed, 1 unresolved discussion (waiting on @jianglai and @weiminyu)

___
*[core/src/main/java/google/registry/beam/common/RegistryPipelineWorkerInitializer.java, line 46 at r3](https://reviewable.io/reviews/google/nomulus/1348#-MksA3LS7o1ZD2K6YvY6:-MkwDgZFBLg8vfjOcVh0:b-896fix) ([raw file](https://github.com/google/nomulus/blob/cf8b8f6e932b1a124033fad7621dbb7debb9e8d7/core/src/main/java/google/registry/beam/common/RegistryPipelineWorkerInitializer.java#L46)):*
<details><summary><i>Previously, weiminyu (Weimin Yu) wrote…</i></summary><blockquote>

> ```
>  == 
> ```
equals() is favored, at least internally: go/java-practices/enums#equals
</blockquote></details>

Done.

File: core/src/main/java/google/registry/reporting/icann/ActivityReportingQueryBuilder.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
   static final String EPP_METRICS = "epp_metrics";
   static final String WHOIS_COUNTS = "whois_counts";
   static final String ACTIVITY_REPORT_AGGREGATION = "activity_report_aggregation";
-  final String BIGQUERY_DATA_SET = tm().isOfy() ? "icann_reporting" : "cloud_sql_icann_reporting";
+  final String bigQueryDataSet = tm().isOfy() ? "icann_reporting" : "cloud_sql_icann_reporting";
```
Comment (CydeWeys) [inline core/src/main/java/google/registry/reporting/icann/ActivityReportingQueryBuilder.java:41]: I added a better fix for this in PR #1361 .

**Review by weiminyu — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1348)* status: 16 of 17 files reviewed, 1 unresolved discussion (waiting on @jianglai)

**Review by jianglai — COMMENTED** [review]
> Reviewed 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/google/nomulus/1348)* status: all files reviewed, 1 unresolved discussion (waiting on @CydeWeys)

___
*[core/src/main/java/google/registry/reporting/icann/ActivityReportingQueryBuilder.java, line 41 at r4](https://reviewable.io/reviews/google/nomulus/1348#-MkmkB74IBFNYXaBEXql-r4-41:-MkwzKhE9jBoExLUL6IH:bp3aow6) ([raw file](https://github.com/google/nomulus/blob/bf6ce77dd9f8ffa6efe70b48e482397b88c324a5/core/src/main/java/google/registry/reporting/icann/ActivityReportingQueryBuilder.java#L41)):*
<details><summary><i>Previously, CydeWeys (Ben McIlwain) wrote…</i></summary><blockquote>

I added a better fix for this in PR #1361 .
</blockquote></details>

Reverted.

**Pull request merged**